File: dmoj/ccc23s1/ccc23s1.py
n = int(input())
arr = list(map(int, input().split()))
brr = list(map(int, input().split()))

ans = 3*(sum(arr)+sum(brr))

for i in range(n-1):
    if arr[i] and arr[i+1]: ans -= 2
    if brr[i] and brr[i+1]: ans -= 2

for i in range(0, n, 2):
    if arr[i] and brr[i]: ans -= 2

print(ans)

File: dmoj/ccc23s1/ccc23s1.cpp
#include <iostream>
using namespace std;

const int MN = 2e5 + 5;
int n;
bool arr[MN], brr[MN];
int ans;

int main(){
    cin.sync_with_stdio(0); cin.tie(0);
    cin >> n;
    for(int i = 0; i < n; i++) cin >> arr[i], ans += 3*arr[i];
    for(int i = 0; i < n; i++) cin >> brr[i], ans += 3*brr[i];
    for(int i = 0; i < n-1; i++){
        ans -= 2*(arr[i] && arr[i+1]);
        ans -= 2*(brr[i] && brr[i+1]);
    }
    for(int i = 0; i < n; i += 2) ans -= 2*(arr[i] && brr[i]);
    cout << ans << '\n';
}
